distribution/test/itests/test-itests-catalog/src/test/java/ddf/catalog/test/TestSingleSignOn.java

```diff
@@ -0,0 +1,261 @@
+        // TODO: Maybe don't go out to the internet for this? The schema itself is going to go out to the internet anyway though.
+        URL schemaFile = new URL(
+                "http://docs.oasis-open.org/security/saml/v2.0/saml-schema-metadata-2.0.xsd");
+        SchemaFactory schemaFactory = SchemaFactory.newInstance(XMLConstants.W3C_XML_SCHEMA_NS_URI);
+        Schema schema = schemaFactory.newSchema(schemaFile);
+        Validator validator = schema.newValidator();
+
+        // Get the metadata
+        String clientMetadata = get(SERVICE_ROOT + "/idp/login/metadata").asString();
+        String serverMetadata = get(SERVICE_ROOT + "/saml/sso/metadata").asString();
+
+        // TODO: The server metadata requires an "index" attribute to validate properly
+        validator.validate(new StreamSource(new StringReader(clientMetadata)));
+        //        validator.validate(new StreamSource(new StringReader(serverMetadata)));
+
+        // To find the right inputs for the settings, go into the metatype.xml file.
+        // The key is the "id" and the value type is determined by the cardinality as such:
+        // Positive = array, negative = vector, 0 (none) = single variable
+        Dictionary<String, Object> clientSettings = new Hashtable<>();
+        clientSettings.put("metadata", clientMetadata);
+        Dictionary<String, Object> serverSettings = new Hashtable<>();
+        serverSettings.put("spMetadata", new String[] {serverMetadata});
+
+        // Update the client and server with the metadata
+        Configuration clientConfig = getAdminConfig()
+                .getConfiguration("org.codice.ddf.security.idp.client.IdpMetadata", null);
+        Configuration serverConfig = getAdminConfig()
+                .getConfiguration("org.codice.ddf.security.idp.server.IdpEndpoint", null);
+        clientConfig.update(clientSettings);
+        serverConfig.update(serverSettings);
+        assertThat(clientConfig.getProperties(), notNullValue());
+        assertThat(serverConfig.getProperties(), notNullValue());
+    }
+
+    private String getRedirectUrl(Response response) {
+        String fullUrl = "";
+
+        // We can either get a legit redirect from the header, or we can have javascript redirect us.
+        // Whenever javascript redirects us, we have to do ugly, fragile parsing of the HTML.
+        if (response.headers().hasHeaderWithName("Location")) {
+            fullUrl = response.header("Location");
+            // TODO: Figure out how to do this using xpath
+        } else if (response.getBody().asString()
+                .contains("<title>Redirect</title>")) { // Javascript redirect
+            Pattern pattern = Pattern.compile("encoded *= *\"(.*)\"", Pattern.CASE_INSENSITIVE);
+            Matcher matcher = pattern.matcher(response.getBody().asString());
+            matcher.find();
+            fullUrl = matcher.group(1);
+        } else {
+            // TODO: this is unexpected. Log an error?
```

Comment:
Should handle this else. Probably want to fail the test here since it cannot be completed at this stage